src/components/Track/index.js

```diff
@@ -43,8 +78,8 @@ export default class Track extends React.Component{
           {
             _.map(segments, segment =>
               <Segment
-                key={segment.slug} trackSlug={slug} name={segment.name} slug={segment.slug} lengthKm={segment.lengthKm}
-                completed={segment.completed} updateSegment={this.updateSegment}
+                key={`${segment.slug}-${this.state.segmentsCompleted[segment.slug]}`} trackSlug={slug} name={segment.name} slug={segment.slug} lengthKm={segment.lengthKm}
```

Comment:
this is quite funky... I guess its weird because I'm tracking the parents state... but shouldn't it be enough to pass the parents state in as props? or am I missing something?